### PR TITLE
feat: upgrade lifecycle events

### PR DESCRIPTION
**Pull request by brunozoric**
## Changes
Add topic names to all lifecycle events.

Change naming in the Headless CMS lifecycle events.

## How Has This Been Tested?
Jest.

### CODE REVIEW

**Review by Pavel910 — APPROVED** [review]

File: packages/api-page-builder/src/graphql/crud/system.crud.ts [hunk 1]
```diff
@@ -25,8 +25,8 @@ export interface CreateSystemCrudParams {
 }
 export const createSystemCrud = (params: CreateSystemCrudParams): SystemCrud => {
     const { context, storageOperations, getTenantId } = params;
-    const onBeforeInstall = createTopic<OnBeforeInstallTopicParams>();
-    const onAfterInstall = createTopic<OnAfterInstallTopicParams>();
+    const onBeforeInstall = createTopic<OnBeforeInstallTopicParams>("pageBuilder.onBeforeInstall");
```
Comment (endymion1818) [inline packages/api-page-builder/src/graphql/crud/system.crud.ts:28]: If we're following `on{conjunction}{noun}{subordinatenoun}{action}`, then we're missing a noun here.

Can I suggest `onBeforeSystemInstall` ?

File: packages/api-headless-cms/src/crud/contentModelGroup/beforeDelete.ts [hunk 2]
```diff
@@ -5,14 +5,14 @@ import { CmsGroupPlugin } from "~/plugins/CmsGroupPlugin";
 import WebinyError from "@webiny/error";
 
 interface AssignBeforeGroupDeleteParams {
-    onBeforeDelete: Topic<BeforeGroupDeleteTopicParams>;
+    onGroupBeforeDelete: Topic<BeforeGroupDeleteTopicParams>;
```
Comment (endymion1818) [inline packages/api-headless-cms/src/crud/contentModelGroup/beforeDelete.ts]: This is much better. But the conjunction is still not inline with the majority of other operations

More explicit never hurt anyone, I'm tempted to suggest `onBeforeContentModelGroupDelete`

File: packages/api-headless-cms/src/crud/system.crud.ts [hunk 3]
```diff
@@ -34,8 +35,14 @@ interface CreateSystemCrudParams {
 export const createSystemCrud = (params: CreateSystemCrudParams): CmsSystemContext => {
     const { getTenant, getLocale, storageOperations, context, getIdentity } = params;
 
-    const onBeforeInstall = createTopic<BeforeInstallTopicParams>();
-    const onAfterInstall = createTopic<AfterInstallTopicParams>();
+    const onSystemBeforeInstall = createTopic<BeforeInstallTopicParams>(
+        "cms.onSystemBeforeInstall"
```
Comment (endymion1818) [inline packages/api-headless-cms/src/crud/system.crud.ts:39]: ~`onBeforeSystemInstall` if we want to stick strictly to the pattern would be better~

File: packages/api-headless-cms/src/types.ts [hunk 4]
```diff
@@ -1413,16 +1458,51 @@ export interface CmsModelContext {
...
+     */
     onBeforeModelDelete: Topic<BeforeModelDeleteTopicParams>;
+    /**
+     * @deprecated
+     */
     onAfterModelDelete: Topic<AfterModelDeleteTopicParams>;
+    /**
+     * Lifecycle events - released in 5.33.0
+     */
+    onModelBeforeCreate: Topic<BeforeModelCreateTopicParams>;
```
Comment (endymion1818) [inline packages/api-headless-cms/src/types.ts:1498]: `onBeforeModelCreate` etc

**Review by endymion1818 — COMMENTED** [review]
Great improvements and really helpful.

I think I commented everywhere, and on Slack I sent a message about a proposed format for these events, which might be helpful: 

`on{conjunction}{noun}{subordinatenoun}{action}`

**Pull request merged**